request-manager/spec/node-check.go

```diff
@@ -0,0 +1,498 @@
+}
+
+/* ========================================================================== */
+type ConditionalHasEqNodeCheck struct{}
+
+/* Conditional nodes must specify `eq`. */
+func (check ConditionalHasEqNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	if node.IsConditional() {
+		if len(node.Eq) == 0 {
+			return MissingValueError{sequenceName, &node.Name, "eq", "required for conditional nodes"}
+		}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type NonconditionalHasTypeNodeCheck struct{}
+
+/* Nonconditional nodes must specify a type. */
+func (check NonconditionalHasTypeNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	if !node.IsConditional() {
+		if node.NodeType == nil {
+			return MissingValueError{sequenceName, &node.Name, "type", "required for nonconditional nodes"}
+		}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type NonconditionalNoIfNodeCheck struct{}
+
+/* Nononditional nodes may not specify `if`. */
+func (check NonconditionalNoIfNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	if !node.IsConditional() {
+		if node.If != nil {
+			return InvalidValueError{sequenceName, &node.Name, "if", []string{*node.If}, "no value; noncoditional nodes may not specify if"}
+		}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type NonconditionalNoEqNodeCheck struct{}
+
+/* Nononditional nodes may not specify `eq`. */
+func (check NonconditionalNoEqNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	if !node.IsConditional() {
+		if len(node.Eq) != 0 {
+			eq := fmt.Sprintf("%v", node.Eq)
+			return InvalidValueError{sequenceName, &node.Name, "eq", []string{eq}, "no value; noncoditional nodes may not specify eq"}
+		}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type RetryIfRetryWaitNodeCheck struct{}
+
+/* If `retryWait` is set, `retry` must be set (nonzero). */
+func (check RetryIfRetryWaitNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	if node.RetryWait != "" && node.Retry == 0 {
+		return MissingValueError{sequenceName, &node.Name, "retry", "required when `retryWait` field set"}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type ValidRetryWaitNodeCheck struct{}
+
+/* `retryWait` should be a valid duration. */
+func (check ValidRetryWaitNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	if node.RetryWait != "" {
+		if _, err := time.ParseDuration(node.RetryWait); err != nil {
+			return InvalidValueError{sequenceName, &node.Name, "retryWait", []string{node.RetryWait}, "valid duration string"}
+		}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type RequiredArgsProvidedNodeCheck struct {
```

Comment:
Can you add a check to make sure that any sequences called actually exist? It should be pretty similar to this one, except at the beginning filter out jobs by checking that node category != job.